client/src/components/GraphViewerComponent/GraphViewerComponent.js

```diff
@@ -467,10 +467,10 @@ class GraphViewerComponent extends React.Component {
     this.setState({ canShowAllRelationships: true });
   }
 
-  onRelationshipCreate = async relationship => {
+  onRelationshipCreate = relationship => {
     if (relationship) {
       this.cyRef.current.addRelationships([ relationship ]);
-      await this.cyRef.current.doLayout();
+      this.cyRef.current.updateNodeColors();
```

Comment:
Would this make more sense to place inside the addRelationship function itself? Or would that cause other issues?

Keeping the color logic internal to the cytoscape component would be cleaner.

Reply:
Done